core/src/main/java/kafka/server/share/ShareFetchUtils.java

```diff
@@ -117,4 +119,17 @@ static long offsetForEarliestTimestamp(TopicIdPartition topicIdPartition, Replic
                 Optional.empty(), true).timestampAndOffsetOpt();
         return timestampAndOffset.isEmpty() ? (long) 0 : timestampAndOffset.get().offset;
     }
+
+    /**
+     * The method is used to get the offset for the latest timestamp for the topic-partition.
+     *
+     * @return The offset for the latest timestamp.
+     */
+    static long offsetForLatestTimestamp(TopicIdPartition topicIdPartition, ReplicaManager replicaManager) {
```

Comment:
// Visible for testing

Reply:
When you are in this file then please write the same for other method.

Reply:
Actually this method is not visible for testing purposes, but because it is being used in a separate file, that's why it is package private. I will mark the method processFetchResponse as // Visible for Testing though